Lab2/scripts/deployment.sh

```diff
@@ -60,12 +71,23 @@ if [[ $server -eq 1 ]]; then
   fi
 
   sam build -t template.yaml --use-container
-  sam deploy --config-file samconfig.toml --region=$REGION
+
+  if [ "$IS_RUNNING_IN_EVENT_ENGINE" = true ]; then
+    sam deploy --config-file samconfig.toml --region=$REGION --parameter-overrides EventEngineParameter=$IS_RUNNING_IN_EVENT_ENGINE AdminUserPoolCallbackURLParameter=$ADMIN_SITE_URL
+  else
+    sam deploy --config-file samconfig.toml --region=$REGION --parameter-overrides EventEngineParameter=$IS_RUNNING_IN_EVENT_ENGINE
+  fi
+  
   cd ../scripts
 fi  
 
-ADMIN_SITE_URL=$(aws cloudformation describe-stacks --stack-name serverless-saas --query "Stacks[0].Outputs[?OutputKey=='AdminAppSite'].OutputValue" --output text)
-LANDING_APP_SITE_URL=$(aws cloudformation describe-stacks --stack-name serverless-saas --query "Stacks[0].Outputs[?OutputKey=='LandingApplicationSite'].OutputValue" --output text)
+if [ "$IS_RUNNING_IN_EVENT_ENGINE" = false ]; then
```

Comment:
can we combine if True code above here?

Reply:
We cannot because at line 76 while deploying shared-template.yaml we need to pass the admin site url and saas application url so that we can configure them as callback urls in our Cognito userpools. So before we run "sam deploy" we need to have admin site and saas application url's and that is the reason we are getting those url's in the above if true condition.